Firestore/core/src/firebase/firestore/model/field_path.h

```diff
@@ -47,10 +48,28 @@ class FieldPath : public impl::BasePath<FieldPath> {
   FieldPath(const IterT begin, const IterT end) : BasePath{begin, end} {
   }
   FieldPath(std::initializer_list<std::string> list) : BasePath{list} {
+    if (size() == 0) {
+      api::ThrowInvalidArgument(
+          "Invalid field path. Provided names must not be empty.");
+    }
+
+    for (size_t i = 0; i < size(); i++) {
+      if ((*this)[i].size() == 0) {
+        api::ThrowInvalidArgument(
+            "Invalid field name at index %s. Field names must not be empty.",
+            i);
+      }
+    }
   }
   explicit FieldPath(SegmentsT&& segments) : BasePath{std::move(segments)} {
   }
 
+  /**
+   * Creates and returns a new path from a dot-separatedfield-path string,
```

Comment:
Nit: no space in-between `dot-separated` and `field-path`.

Reply:
woops, good catch.